test/y2partitioner/dialogs/lvm_lv_size_test.rb

```diff
@@ -0,0 +1,222 @@
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com
+
+require_relative "../test_helper"
+
+require "cwm/rspec"
+require "y2storage"
+require "y2partitioner/dialogs"
+require "y2partitioner/sequences/controllers"
+
+describe Y2Partitioner::Dialogs::LvmLvSize do
+  using Y2Storage::Refinements::SizeCasts
+
+  before do
+    devicegraph_stub("lvm-two-vgs.yml")
+  end
+
+  subject { described_class.new(controller) }
+
+  let(:controller) { Y2Partitioner::Sequences::Controllers::LvmLv.new(vg) }
+
+  let(:vg) { Y2Storage::LvmVg.find_by_vg_name(current_graph, "vg0") }
+
+  let(:current_graph) { Y2Partitioner::DeviceGraphs.instance.current }
+
+  include_examples "CWM::Dialog"
+
+  describe Y2Partitioner::Dialogs::LvmLvSize::SizeWidget do
+    subject(:widget) { described_class.new(controller) }
+
+    include_examples "CWM::CustomWidget"
+
+    describe "#store" do
+      before do
+        allow(widget).to receive(:value).and_return(value)
+
+        allow(Y2Partitioner::Dialogs::LvmLvSize::MaxSizeDummy)
+          .to receive(:new).and_return(max_size_widget)
+
+        allow(Y2Partitioner::Dialogs::LvmLvSize::CustomSizeInput)
+          .to receive(:new).and_return(custom_size_widget)
+      end
+
+      let(:max_size_widget) { double("MaxSizeDummy", size: max_size) }
+
+      let(:custom_size_widget) { double("CustomSizeInput", size: custom_size) }
+
+      let(:max_size) { nil }
+
+      let(:custom_size) { nil }
+
+      let(:value) { :custom_size }
+
+      it "sets #size_choice in the controller" do
+        expect(controller.size_choice).to be_nil
+        widget.store
+        expect(controller.size_choice).to eq(value)
+      end
+
+      context "when max size is selected" do
+        let(:value) { :max_size }
+
+        let(:max_size) { 10.GiB }
+
+        it "sets #size in the controller to max possible size" do
+          widget.store
+          expect(controller.size).to eq(max_size)
+        end
+      end
+
+      context "when custom size is selected" do
+        let(:value) { :custom_size }
+
+        let(:custom_size) { 1.GiB }
+
+        it "sets #size in the controller to the value indicated" do
+          widget.store
+          expect(controller.size).to eq(custom_size)
+        end
+      end
+    end
+  end
+
+  describe Y2Partitioner::Dialogs::LvmLvSize::MaxSizeDummy do
+    subject(:widget) { described_class.new(1.GiB) }
+
+    include_examples "CWM::AbstractWidget"
+  end
+
+  describe Y2Partitioner::Dialogs::LvmLvSize::CustomSizeInput do
+    subject(:widget) { described_class.new(5.GiB, 1.GiB, 10.GiB) }
+
+    before do
+      allow(widget).to receive(:value).and_return(value)
+    end
+
+    let(:value) { nil }
+
+    include_examples "CWM::AbstractWidget"
+
+    describe "#validate" do
+      context "when indicated value in not a valid size" do
```

Comment:
really-np: I would change `indicated` by `given` (it is a common term when writing tests).